transmission.go

```diff
@@ -146,6 +148,14 @@ func (b *batch) sendRequest(e *Event) {
 		start = b.testNower.Now()
 	}
 	timestamp := e.Timestamp
+
+	url, err := url.Parse(e.APIHost)
+	if err != nil {
+		// TODO add logging or something to raise this error
+		sd.Increment("url_parse_errors")
+		return
```

Comment:
I think this error needs to get sent down the response channel. (which means I also think line 163 below should be sending the error down the responses channel.) 
It looks safe to me to bring the creation of the Response{} and the defer to send it down the responses channel up to the beginning of this function then both here and lien 163 can set the error on the response.